fdbserver/RestoreMaster.actor.cpp

```diff
@@ -641,7 +641,8 @@ ACTOR static Future<Standalone<VectorRef<RestoreRequest>>> collectRestoreRequest
 ACTOR static Future<Version> collectBackupFiles(Reference<IBackupContainer> bc, std::vector<RestoreFileFR>* rangeFiles,
                                                 std::vector<RestoreFileFR>* logFiles, Database cx,
                                                 RestoreRequest request) {
-	state BackupDescription desc = wait(bc->describePartitionedBackup());
+	state bool partitioned = wait(bc->isPartitionedBackup());
```

Comment:
Can we use a knob to skip `isPartitionedBackup()` in prod and set it by DBA manually? 
Since this function lists all log files, it can take some time (minutes) for a large backup.

For example, if knob = 0, use isPartitionedBackup(); if knob = 1, set partitioned = 1; if knob = 2, set partitioned = 0. 
Add comment to the knob to explain the meaning of each value.

In simulation, we can always set the knob to 0.

Reply:
I think a knob is not doable, since it's a fixed value and a container can be either old or new backup.

I agree listing all log files is expensive. For simulation, it's ok. We need some metadata in the backup container to tell us which style of backup is used. In that way, this can be a much cheaper function.